Automattic/woocommerce-payments
# Add payment activity widget emoji survey

Fixes #8485

> [!NOTE]  
> Based on @dpaun1985's PR #8309 – commits have been cherry-picked to this branch.

#### Changes proposed in this Pull Request

Emoji-based survey component to capture the user sentiment for the new Payments Overview Widget.

The survey accepts 3 fields:
* `rating`: 'very-unhappy', 'unhappy', 'neutral', 'happy', 'very-happy'
* `comments`: user-defined text
* `wcpay-version`: WooPayments plugin version as text (e.g. `7.4.0`)

![image](https://github.com/Automattic/woocommerce-payments/assets/1288616/ced1af30-a6a8-4b96-b362-62848ff53729)

**iOS screenshots**
<div>
<img width="32%" alt="emoji-survey" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/27507e48-3e3a-4c3a-a250-fb6f6c342045">
<img width="32%" alt="emoji-survey-comment" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/beb88ffd-040c-4fdc-a158-ba0347e162c1">
<img width="32%" alt="emoji-survey-comment" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/218afef1-1572-4b17-8d0c-9b7c076df347" >
</div> 




#### Todo
- [x] Place within the Payments Activity Widget introduced in #8412
- [x] Use common component for close button
- [x] Check against latest MVP design changes
- [x] Fix / update tests
- [x] Add WooPayments major version number as form variable
- [x] Check changes to WPCOM survey endpoint
- [x] Deploy changes to WPCOM survey endpoint #8314


#### Testing instructions
- run `wp option update _wcpay_feature_payment_overview_widget 1` in CLI (`npm run cli` if docker)
- change your host file to point `public-api.wordpress.com` and `dev-mc.a8c.com` to your sandbox's IP address if you didn't already
- log-in to your sandbox
- apply the patch created in #8314 to your dev server (`arc patch D140691`)
- in the client admin panel, go to `Payments -> Overview`
- by clicking the first 3 emoji, a textarea for comments should appear
- for the highest rates ( last 2 emojis ) the survey will be sent without comments
- after submitting the results, go to https://dev-mc.a8c.com/marketing/survey/index.php?survey=wcpay-payments-overview and check the answers.
- to re-submit, run `wp option delete wcpay_survey_payment_overview_submitted` to clear your local submission state

https://github.com/Automattic/woocommerce-payments/assets/82826872/71aaf433-8d90-49c9-9e15-a8368aa64707


-------------------


**Post merge**
